File: dener8/programmers_숫자의표현.py
def solution(n):
    answer = 0
    numbers = [i for i in range(1, 10001)]
    for i in range(len(numbers)):
        temp_sum = numbers[i]
        for j in range(i + 1, len(numbers) + 1):
            if temp_sum == n:
                answer += 1
                # print(numbers[i : j])
                break
            elif temp_sum > n:
                break
            temp_sum += numbers[j]
    return answer

File: myunghyunnero/programmers_숫자의표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    int start = 1;
    int end=1;
    int sum=0;
    for(int i=1;i<=n;i++){
        sum+=end;
        end++;
        if(sum==n){
            answer++;
        }else if(sum>n){
            for(int j=start;j<end;j++){
                sum-=start;
                start++;
                if(sum==n){answer++;}
                if(sum<n){
                    break;
                }
            }
        }
    }
    
    return answer;
}